crates/worker/src/partition/state_machine/command_interpreter/mod.rs

```diff
@@ -287,6 +294,47 @@ where
             }
         }
 
+        if service_invocation.invocation_target.invocation_target_ty()
+            == InvocationTargetType::Workflow(WorkflowHandlerType::Workflow)
+        {
+            let keyed_service_id = service_invocation
+                .invocation_target
+                .as_keyed_service_id()
+                .expect("When the handler type is Workflow, the invocation target must have a key");
+
+            let service_status = state.get_virtual_object_status(&keyed_service_id).await?;
+
+            // If locked, then we check the original invocation
+            if let VirtualObjectStatus::Locked(original_invocation_id) = service_status {
+                if let Some(response_sink) = service_invocation.response_sink {
+                    let invocation_status =
+                        state.get_invocation_status(&original_invocation_id).await?;
+
+                    match invocation_status {
+                        InvocationStatus::Completed(
+                            CompletedInvocation { response_result, .. }) => {
+                            self.send_response_to_sinks(
+                                effects,
+                                &service_invocation.invocation_id,
+                                None,
+                                iter::once(response_sink),
+                                response_result,
+                            );
+                        }
+                        InvocationStatus::Free => panic!("Unexpected state, the InvocationStatus cannot be Free for invocation {} given it's in locked status", original_invocation_id),
+                        is => effects.append_response_sink(
+                            original_invocation_id,
+                            is,
+                            response_sink
+                        )
+                    }
+                }
+
+                // TODO ADD ATTACH INVOCATION ID NOTIFICATION TO INGRESS!!!!!!
+                return Ok(());
+            }
+        }
```

Comment:
After an offline chat with Francesco: Workflows don't need to use the idempotency table because calls to a workflow are uniquely identified through the service key. So we add a bit of special casing for this specific scenario.